backend/windmill-queue/src/jobs.rs

```diff
@@ -382,50 +383,109 @@ pub async fn add_completed_job<R: rsmq_async::RsmqConnection + Clone + Send>(
+        if is_global {
+            SUPERADMIN_SECRET_EMAIL.to_string()
+        } else {
+            "g/error_handler".to_string()
+        },
+        None,
+        None,
+        Some(job_id),
+        Some(job_id),
+        None,
+        false,
+        false,
+        None,
+        true,
+        tag,
+    )
+    .await?;
+    tx.commit().await?;
+
+    let error_handler_type = if is_global { "global" } else { "workspace" };
+    tracing::info!(
+        "Sent error of job {job_id} to {error_handler_type} error handler under uuid {uuid}"
+    );
+
+    Ok(())
+}
+
 pub async fn send_error_to_global_handler<R: rsmq_async::RsmqConnection + Clone + Send>(
     rsmq: Option<R>,
     queued_job: &QueuedJob,
     db: &Pool<Postgres>,
     result: &serde_json::Value,
 ) -> Result<(), Error> {
-    if let Some(ref global_error_handler) = *GLOBAL_ERROR_HANDLER_PATH_IN_ADMINS_WORKSPACE {
-        let w_id = &queued_job.workspace_id;
-        let job_id = queued_job.id;
-        let mut tx: QueueTransaction<'_, _> = (rsmq, db.begin().await?).into();
-        let (job_payload, tag) =
-            script_path_to_payload(&global_error_handler, tx.transaction_mut(), "admins").await?;
-        let mut args = result.as_object().unwrap().clone();
-        args.insert("workspace_id".to_string(), json!(w_id));
-        args.insert("job_id".to_string(), json!(job_id));
-        args.insert("path".to_string(), json!(queued_job.script_path));
-        args.insert("is_flow".to_string(), json!(queued_job.raw_flow.is_some()));
-        args.insert("email".to_string(), json!(queued_job.email));
-
-        let (uuid, tx) = push(
-            tx,
-            "admins",
-            job_payload,
-            args,
-            "global",
-            SUPERADMIN_SECRET_EMAIL,
-            SUPERADMIN_SECRET_EMAIL.to_string(),
-            None,
-            None,
-            Some(job_id),
-            Some(job_id),
-            None,
-            false,
+    let w_id = &queued_job.workspace_id;
+    let mut tx = db.begin().await?;
+    let error_handler = sqlx::query_scalar!(
+        "SELECT error_handler FROM workspace_settings WHERE workspace_id = $1",
+        &w_id
+    )
+    .fetch_optional(&mut tx)
+    .await
+    .context("sending error to global handler")?
+    .ok_or_else(|| Error::InternalErr(format!("no workspace settings for id {w_id}")))?;
+
+    if let Some(error_handler) = error_handler {
+        run_error_handler(
+            rsmq.clone(),
+            queued_job,
+            db,
+            result,
+            &error_handler.strip_prefix("script/").unwrap(),
             false,
-            None,
+        )
+        .await?;
+    }
+
+    if let Some(ref global_error_handler) = *GLOBAL_ERROR_HANDLER_PATH_IN_ADMINS_WORKSPACE {
+        run_error_handler(
+            rsmq.clone(),
+            queued_job,
+            db,
+            result,
+            global_error_handler,
             true,
-            tag,
         )
         .await?;
```

Comment:
If you do this, failure of error handler will prevent global error handler to trigger :)

Reply:
You want to run both and return a new meta error if error that is merge or one of them